src/libraries/System.Private.CoreLib/src/System/IO/Path.cs

```diff
@@ -926,5 +926,95 @@ private static string GetRelativePath(string relativeTo, string path, StringComp
         /// Returns true if the path ends in a directory separator.
         /// </summary>
         public static bool EndsInDirectorySeparator(string path) => PathInternal.EndsInDirectorySeparator(path);
+
+        /// <summary>
+        /// Removes any redundant segments found in the specified path.
+        /// </summary>
+        /// <param name="path">A string containing an absolute or relative path that may or may not contain redundant segments.</param>
+        /// <returns><paramref name="path" /> without any redundant segments.</returns>
+        [return: NotNullIfNotNull("path")]
+        public static string? RemoveRedundantSegments(string? path)
+        {
+            if (path == null)
+                return null;
+
+            if (PathInternal.IsEffectivelyEmpty(path))
+                return string.Empty;
+
+            ValueStringBuilder sb = new ValueStringBuilder(path.Length);
+            ReadOnlySpan<char> pathRoot = GetPathRoot(path);
+
+            if (!PathInternal.TryRemoveRedundantSegments(path, pathRoot.Length, ref sb))
+            {
+                sb.Dispose();
+                return path;
+            }
+
+            string output = sb.ToString();
+            sb.Dispose();
```

Comment:
`ValueStringBuilder.ToString` does `Dispose`. You do not need to do it second time.

Reply:
Oh that's right. I'll remove it.